src/main/webapp/scripts/map.js

```diff
@@ -80,12 +88,79 @@ window.initMap = function () {
         strokeOpacity: 0.8,
         strokeWeight: 2
       });
+      // Once cards are retrieved...
+      // Add vendor's markers
+      addVendorsToMap(map);
     },
-    (error) => {
-      if (error.code == error.PERMISSION_DENIED)
-        alert('Permission denied to access location');
-    });
+      (error) => {
+        if (error.code == error.PERMISSION_DENIED)
+          alert('Permission denied to access location');
+      });
   } else {
     alert('This browser does not support location');
   }
+};
+
+/**
+ * Function that adds vendor marks to the map
+ */
+const addVendorsToMap = (map) => {
+  const salecardTemplate = document.getElementById('salecard-template');
+  const salecardsContainer = document.getElementById('salecards-container');
+
+  //Clean previously retrieved cards
+  salecardsContainer.innerHTML = '';
+
+  //Display number of vendor found
+  displayNumberOfVendors(vendors);
+
+  Object.keys(vendors).forEach(vendorNumber => {
+    let vendor = vendors[vendorNumber];
+    let salecard = vendor.saleCard;
+
+    createModal(vendor, salecard, salecardTemplate, salecardsContainer);
+
+    var marker = new google.maps.Marker({
+      map: map,
+      position: {
+        lat: salecard.location.salepoint.lat,
+        lng: salecard.location.salepoint.lng
+      },
+      title: salecard.businessName
+    });
+
+    // Toggles salecard's modal when vendor's marker is clicked
+    marker.addListener('click', () => {
+      $(`#modal${vendor.id}`).modal('toggle');
+    })
+  });
+};
+
+/**
+ * Function that creates a modal given a salecard
+ */
+const createModal = (vendor, salecard, salecardTemplate, salecardsContainer) => {
```

Comment:
Also add JSDoc here

Reply:
Here is more clear because `salecardTemplate` and `salecardsContainer` are HTML elements.

Reply:
Yep, I think `Element` is valid in JSDoc.

Reply:
+1 please add JSDoc